server/src/main/java/io/druid/segment/realtime/appenderator/AppenderatorImpl.java

```diff
@@ -576,9 +595,14 @@ private DataSegment mergeAndPush(final SegmentIdentifier identifier, final Sink
           tuningConfig.getIndexSpec()
       );
 
-      DataSegment segment = dataSegmentPusher.push(
-          mergedFile,
-          sink.getSegment().withDimensions(IndexMerger.getMergedDimensionsFromQueryableIndexes(indexes))
+      // Retry pushing segments because uploading to deep storage might fail especially for cloud storage types
+      final DataSegment segment = RetryUtils.retry(
```

Comment:
It makes sense and sounds even better. I think its better to make the pushers to retry by themselves in a follow-up pr because there are somethings to do like checking the ```S3RETRY``` predicate can be applied to other types of pushers. Maybe it's possible to make the retrying smarter. What do you think?